File: backend/src/stolonet/api/dto/telemetry.py
from datetime import datetime

from pydantic import BaseModel

from stolonet.domain.enums import MetricType


class TimestampedReadingResponse(BaseModel):
    node_id: str
    metric: MetricType
    value: float
    unit: str
    timestamp: datetime


class MetricAverageResponse(BaseModel):
    node_id: str
    metric: MetricType
    average_value: float | None
    unit: str
    hours: int

File: frontend/src/interfaces/telemetry.ts
import type {MetricType} from "../enums/metric_type.ts";

export interface Telemetry {
  node_id: string;
  metric: MetricType;
  value: number;
  unit: string;
  timestamp: string;
}

export interface MetricAverage {
  node_id: string;
  metric: MetricType;
  average_value: number | null;
  unit: string;
  hours: number;
}
